### PR TITLE
Fix/profile_user_context

### PR DESCRIPTION
**Pull request by TanyaDovzhenko**

### CODE REVIEW

File: src/components/common/header/HeaderInfo.tsx [hunk 1]
```diff
@@ -1,49 +1,47 @@
...
+    const logout = () => {
+        localStorage.removeItem('token')
+        navigate('/sign-in')
+    }
 
     return (
-        <div className={style.profile_container}>
-            {authorized
-                ? <Button onClick={() => navigate('/profile')}>
+        <div> {user ?
```
Comment (RamboUA) [inline src/components/common/header/HeaderInfo.tsx:22]: As it is a separate div in a result - better to write it from a new line

File: src/components/common/header/HeaderInfo.tsx [hunk 2]
```diff
@@ -1,49 +1,47 @@
...
+                    <img src={logoutIcon} />
+                </Button>
+            </div>
+            : <div>
+                <Button variant='text' size='small' onClick={logout}
+                    sx={{ color: '#fff', textTransform: 'capitalize' }}>
+                    Sing in
+                </Button>
+                <Button variant='text' size='small'
+                    onClick={() => { localStorage.removeItem('token'); navigate('/sign-up') }}
```
Comment (RamboUA) [inline src/components/common/header/HeaderInfo.tsx:40]: Please put this handler to a function outside JSX.

File: src/components/profile/profile-activate/ProfileActivate.tsx [hunk 3]
```diff
@@ -132,21 +126,18 @@ export const ProfileActivate: FC = () => {
                   name='document'
                   onChange={(event) => {
```
Comment (RamboUA) [inline src/components/profile/profile-activate/ProfileActivate.tsx:127]: This handler should be outside as well

**Review by RamboUA — CHANGES_REQUESTED** [review]
Thank you, good job. Please check the comments.
Also check your changes if you have some blank lines in JSX (consider adding it as a eslint rule).
Check handlers in props of component. It's always better to create handler function outside and pass it in props, except some really simple and compact callbacks.

File: src/components/profile/profile-activate/ProfileActivate.tsx (hunk 3 above)
Comment (TanyaDovzhenko) [reply src/components/profile/profile-activate/ProfileActivate.tsx:127]: > This handler should be outside as well

this callback is inside the formic field "{({ errors, touched, isValid, dirty, setFieldValue }) => {" and uses the setFieldValue method, so if it is moved to a separate function, the setFieldValue method will be unavailable

File: src/components/auth/sign-in-form.tsx [hunk 4]
```diff
@@ -21,29 +23,29 @@ const validationSchema = yup.object({
...
 
   const navigate = useNavigate();
 
   const signIn = async (values: FormikValues) => {
-    const res = await userService
-      .signIn({
-        email: values.email,
-        password: values.password,
-      })
+    await userService
```
Comment (RamboUA) [inline src/components/auth/sign-in-form.tsx:33]: Probably it's possible to avoid await here as well

File: src/components/auth/sign-up-form.tsx [hunk 5]
```diff
@@ -41,123 +43,113 @@ const validationSchema = yup.object({
...
-                  flexDirection: 'column',
-                }}
+      <Box sx={formStyles}>
+        <Typography sx={{ textAlign: 'center', fontSize: '2rem' }}>
+          Sign Up
+        </Typography>
+        <Formik
+          initialValues={initialValues}
+          validationSchema={validationSchema}
+          onSubmit={async (values, formikHelpers) => {
```
Comment (RamboUA) [inline src/components/auth/sign-up-form.tsx:76]: I'd recommend to move it outside from JSX

File: src/components/common/header/HeaderInfo.tsx [hunk 6]
```diff
@@ -1,46 +1,51 @@
...
+        localStorage.removeItem('token')
+        navigate('/sign-up')
+    }
 
     return (
-        <div className={style.profile_container}>
-            {authorized
-                ? <Button onClick={() => navigate('/profile')}>
+        <div>
+            {user ? <div>
```
Comment (RamboUA) [inline src/components/common/header/HeaderInfo.tsx:28]: indents are wrong, I would recommend to follow structure:
```
{user ?
   (
      <div>
         ...
      </div>
   ) : (
      <div>
         ...
      </div>
   )
}

```

File: src/components/orders/UpdateOrder/UpdateOrder.tsx [hunk 7]
```diff
@@ -41,9 +42,9 @@ const validationSchema =  yup.object({
 
 const UpdateOrder: FC<IProps> = ({ open, handleClose, order, setOrder }) => {
 
-  const [error,setError] = useState<string>('');
+  const [error, setError] = useState<string>('');
```
Comment (RamboUA) [inline src/components/orders/UpdateOrder/UpdateOrder.tsx:45]: I think it's not necessary to use typing, it should set automatically, when you define error in state as an empty string.
```suggestion
  const [error, setError] = useState('');
```

File: src/components/orders/UpdateOrder/UpdateOrder.tsx [hunk 8]
```diff
@@ -41,9 +42,9 @@ const validationSchema =  yup.object({
...
 
-  const onSubmit = async (values: FormikValues,formikHelpers:any) => {
+  const onSubmit = async (values: FormikValues, formikHelpers: any) => {
```
Comment (RamboUA) [inline src/components/orders/UpdateOrder/UpdateOrder.tsx:47]: Can we try to get rid of any here, please?

File: src/components/orders/UpdateOrder/UpdateOrder.tsx [hunk 9]
```diff
@@ -59,7 +60,7 @@ const UpdateOrder: FC<IProps> = ({ open, handleClose, order, setOrder }) => {
         handleClose();
         formikHelpers.resetForm();
       }
-    }catch (e:any) {
+    } catch (e: any) {
```
Comment (RamboUA) [inline src/components/orders/UpdateOrder/UpdateOrder.tsx:63]: and here

File: src/components/orders/UpdateOrder/UpdateOrder.tsx [hunk 10]
```diff
@@ -74,8 +75,8 @@ const UpdateOrder: FC<IProps> = ({ open, handleClose, order, setOrder }) => {
           await onSubmit(values, formikHelpers);
         }}
       >
-        {({ values, errors, touched, isValid, dirty, setFieldValue }) => {
-          return(
+        {({ errors, touched, isValid, dirty, setFieldValue }) => {
```
Comment (RamboUA) [inline src/components/orders/UpdateOrder/UpdateOrder.tsx:78]:
```suggestion
        {({ errors, touched, isValid, dirty, setFieldValue }) => (
           <Form className={styles.form}>
           ...
```

File: src/components/profile/profile-activate/ProfileActivate.tsx (hunk 3 above)
Comment (RamboUA) [reply src/components/profile/profile-activate/ProfileActivate.tsx:126]: I would not agree with you. You can always pass it as an argument to external function.

**Review by RamboUA — CHANGES_REQUESTED** [review]
Thank you, great, just check please other comments. And one answer to your argument.

**Review by BillyAleks — APPROVED** [review]

**Pull request merged**